staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -819,6 +820,12 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 		if _, hasExcludeBalancerLabel := prevNode.ObjectMeta.Labels[v1.LabelNodeExcludeBalancers]; hasExcludeBalancerLabel {
 			az.excludeLoadBalancerNodes.Delete(prevNode.ObjectMeta.Name)
 		}
+
+		// If not in ready state and not a newly created node, remove from
+		// excludeLoadBalancerNodes cache.
+		if !isNodeReady(prevNode) && cloud.GetCloudTaint(prevNode.Spec.Taints) == nil {
```

Comment:
If I have a newNode, why should the previousNode determine the state of inclusion and exclusion in a load balancer?  It seems as though the only time prevNode should be authoritative is in cases when newNode isn't present, which I believe reduces to deletion use-cases.